meal_mate/app/src/main/java/com/android/engineer/mealmate/data/repository/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.android.engineer.mealmate.data.repository
+
+import com.android.engineer.mealmate.data.remote.model.request.RegisterRequest
+import com.android.engineer.mealmate.data.remote.model.response.RegisterResponse
+import com.android.engineer.mealmate.data.remote.MealAPI
+import com.android.engineer.mealmate.data.utils.API_KEY_VALUE
+import com.android.engineer.mealmate.domain.repository.AuthRepository
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class AuthRepositoryImpl @Inject constructor(
+    private val api: MealAPI
+): AuthRepository {
+
+    override suspend fun registerAccount(registerRequest: RegisterRequest): Flow<RegisterResponse> {
+        return flow {
```

Comment:
@Siduchels  Any idea how do we catch errors here? that was one of my review comment and i don't know the right way to do ...

Reply:
@nasreekar I tried, but it’s not working as expected, so I need to investigate further.

Reply:
@Siduchels  let me know if you find any solution... (may be even after the challenge)